unitTests/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacetTest.ts

```diff
@@ -290,6 +298,51 @@ export function DynamicHierarchicalFacetTest() {
       expect(test.cmp.values.render).toHaveBeenCalled();
     });
 
+    it(`when "enableFacetSearch" option is false
+    it should not render the search element`, () => {
+      options.enableFacetSearch = false;
+      initializeComponent();
+      test.cmp.ensureDom();
+
+      expect(searchFeatureActive()).toBe(false);
+    });
+
+    it(`when "enableFacetSearch" option is false
+    it should not throw when collapsing`, () => {
+      options.enableFacetSearch = false;
+      initializeComponent();
+      test.cmp.ensureDom();
+
+      expect(() => test.cmp.collapse()).not.toThrow();
+    });
+
+    it(`when "enableFacetSearch" option is true
+    it should render the search element`, () => {
+      options.enableFacetSearch = true;
+      initializeComponent();
+      test.cmp.ensureDom();
+
+      expect(searchFeatureActive()).toBe(true);
+    });
+
+    it(`when "enableFacetSearch" option is "undefined" and "moreValuesAvailable" is "true"
+    it should show the search`, () => {
+      initializeComponent();
+      test.cmp.moreValuesAvailable = true;
+      test.cmp.ensureDom();
+
+      expect(searchFeatureDisplayed()).toBe(true);
+    });
+
+    it(`when "enableFacetSearch" option is "undefined" and "moreValuesAvailable" is "true"
+    it should hide the search`, () => {
+      initializeComponent();
+      test.cmp.moreValuesAvailable = false;
+      test.cmp.ensureDom();
+
+      expect(searchFeatureDisplayed()).toBe(false);
+    });
```

Comment:
Those tests are almost an exact copy of [`DynamicFacet`](https://github.com/coveo/search-ui/blob/master/unitTests/ui/DynamicFacet/DynamicFacetTest.ts#L519)'s.